### PR TITLE
AWS with Terraform module

### PR DESCRIPTION
**Pull request by edif2008**
This PR contains the Terraform module that will help the user to create the Lambda function, along with an API Gateway and a IAM role that can be used as a webhook without other configurations. The readme with instructions for deploying the webhook with Terraform is included.

### CODE REVIEW

File: deploy/aws-lambda/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## Deploy in a AWS Lambda Function
+
+You can deploy the webhook to an AWS Lambda function and activate it in your Kubernetes cluster by adding the following module to your Terraform file:
```
Comment (florisvdg) [inline deploy/aws-lambda/README.md]:
```suggestion
You can deploy the webhook to AWS Lambda and activate it in your Kubernetes cluster by adding the following module to your Terraform project: 
```

File: deploy/aws-lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+## Deploy in a AWS Lambda Function
+
+You can deploy the webhook to an AWS Lambda function and activate it in your Kubernetes cluster by adding the following module to your Terraform file: 
+
+```terraform
+module "secrethub_mutating_webhook" {
+  source = "github.com/secrethub/secrethub-kubernetes-mutating-webhook/deploy/aws-lambda"
```
Comment (florisvdg) [inline deploy/aws-lambda/README.md]: We should version this. I think you can do this by appending `?ref=v0.2.0`.

File: deploy/aws-lambda/webhook.tf [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_api_gateway_rest_api" "api" {
+  name = "SecretHubWebhookApi"
```
Comment (florisvdg) [inline deploy/aws-lambda/webhook.tf]: Cosmetic thing:
```suggestion
  name = "SecretHubWebhookAPI"
```

File: deploy/aws-lambda/webhook.tf [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+resource "aws_api_gateway_deployment" "webhook_deploy" {
+  depends_on = [
+    aws_api_gateway_integration.lambda_integration,
+    aws_api_gateway_integration.lambda_root,
+  ]
+
+  rest_api_id = aws_api_gateway_rest_api.api.id
+  stage_name  = "v1"
+
```
Comment (florisvdg) [inline deploy/aws-lambda/webhook.tf]: Some redundant whitespace:
```suggestion
```

File: deploy/aws-lambda/webhook.tf [hunk 5]
```diff
@@ -0,0 +1,116 @@
+data "github_release" "webhook" {
+  repository  = "secrethub-kubernetes-mutating-webhook"
+  owner       = "secrethub"
+  retrieve_by = "latest"
```
Comment (florisvdg) [inline deploy/aws-lambda/webhook.tf]: We should source this from a `variable` and pin it to `v0.2.0`.

File: deploy/aws-lambda/README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+}
+```
```
Comment (florisvdg) [inline deploy/aws-lambda/README.md:9]: We should mention that it depends on the AWS and Kubernetes providers:

````suggestion
```

This module requires the [AWS provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs) and [Kubernetes provider](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs) to be configured in your root project.
````

File: deploy/aws-lambda/README.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+## Deploy in a AWS Lambda Function
```
Comment (florisvdg) [inline deploy/aws-lambda/README.md]:
```suggestion
# Deploy to AWS Lambda
```

File: deploy/aws-lambda/README.md (hunk 2 above)
Comment (edif2008) [reply deploy/aws-lambda/README.md]: Added the version

File: deploy/aws-lambda/webhook.tf (hunk 5 above)
Comment (edif2008) [reply deploy/aws-lambda/webhook.tf]: Adjusted to using tag version

Comment (florisvdg) [reply deploy/aws-lambda/webhook.tf]: Where? Did you push the change?

Comment (edif2008) [reply deploy/aws-lambda/webhook.tf]: I forgot. My bad. now it's pushed

File: deploy/aws-lambda/README.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Deploy to AWS Lambda
+
+You can deploy the webhook to AWS Lambda and activate it in your Kubernetes cluster by adding the following module to your Terraform project: 
+
+```terraform
+module "secrethub_mutating_webhook" {
+  source = "github.com/secrethub/secrethub-kubernetes-mutating-webhook?ref=v0.2.0/deploy/aws-lambda"
```
Comment (florisvdg) [inline deploy/aws-lambda/README.md]: Does this work? 🤔
First the `ref` query parameter and then the local path?

File: deploy/aws-lambda/variables.tf [hunk 9]
```diff
@@ -0,0 +1,9 @@
+variable "secrethub_mutating_webhook_version" {
+  type = string
+  default = "v0.2.0"
```
Comment (florisvdg) [inline deploy/aws-lambda/variables.tf]: Could use a description:
```suggestion
  description = "Version of the SecretHub webhook to deploy."
  default     = "v0.2.0"
```

File: deploy/aws-lambda/README.md (hunk 8 above)
Comment (florisvdg) [reply deploy/aws-lambda/README.md]: I'd expect:
```suggestion
  source = "github.com/secrethub/secrethub-kubernetes-mutating-webhook//deploy/aws-lambda?ref=v0.2.0"
```
(didn't test this)

Comment (edif2008) [reply deploy/aws-lambda/README.md]: Yes, indeed it works just like it is presented.
The one I tried was the following and it didn't work:
`github.com/secrethub/secrethub-kubernetes-mutating-webhook/deploy/aws-lambda?ref=v0.2.0`
Therefore I found the one that is implemented.

Interesting enough, your suggestion works, which I think it is indeed cleaner. I will apply your version.

Comment (edif2008) [reply deploy/aws-lambda/README.md]: Fixed.

File: deploy/aws-lambda/variables.tf (hunk 9 above)
Comment (edif2008) [reply deploy/aws-lambda/variables.tf]: Solved

**Review by florisvdg — APPROVED** [review]
Made some final adjustments 🚀

**Pull request merged**